util-jvm/src/main/scala/com/twitter/jvm/Hotspot.scala

```diff
@@ -56,10 +57,18 @@ class Hotspot extends Jvm {
   private[this] def long(c: Counter) = c.getValue().asInstanceOf[Long]
 
   private[this] def counters(pat: String) = {
-    val cs = jvm.getInternalCounters(pat).asScala
-    cs.map { c =>
-      c.getName() -> c
-    }.toMap
+    val cs = Try(jvm.getInternalCounters(pat).asScala)
```

Comment:
instead of failing now, how about instead failing on construction of the class?  is it important to you that you get the rest of the metrics?

if you want the rest of the Hotspot metrics, this approach broadly seems fine, but I think we probably want to use try / catch so we can shave the allocations that come with using the Try monad

Reply:
I am fine rewriting this to try/catch to reduce overhead.

Reply:
Rewritten using try/catch